recipes/fmilib/all/conanfile.py

```diff
@@ -0,0 +1,145 @@
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["FMILIB_EXTERNAL_LIBS"] = True
+        tc.variables["FMILIB_BUILD_STATIC_LIB"] = not self.options.shared
+        tc.variables["FMILIB_BUILD_SHARED_LIB"] = self.options.shared
+        tc.variables["FMILIB_BUILD_TESTS"] = self.options.with_fmus
+        tc.variables["FMILIB_FMI_STANDARD_HEADERS"] = posixpath.join(self.source_folder, "src", "fmis").replace("\\", "/")
+        tc.variables["FMILIB_GENERATE_DOXYGEN_DOC"] = False
+
+        if not self.options.shared and not self.settings.os in ["Windows", "Macos"]:
+            tc.variables["FMILIB_BUILD_FOR_SHARED_LIBS"] = self.options.get_safe("fPIC", False)
+
+        if is_msvc(self):
+            tc.variables["FMILIB_BUILD_WITH_STATIC_RTLIB"] = is_msvc_static_runtime(self)
+
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.set_property("expat", "cmake_file_name", "EXPAT")
+        deps.set_property("minizip", "cmake_file_name", "MiniZip")
+        deps.set_property("zlib", "cmake_file_name", "ZLIB")
+
+        deps.set_property("expat", "cmake_target_name", "expat")
+        deps.set_property("minizip", "cmake_target_name", "minizip")
+        deps.set_property("zlib", "cmake_target_name", "zlib")
+        deps.generate()
+
+        # This is needed if with_fmus=True
+        vre = VirtualRunEnv(self)
+        vre.generate(scope="build")
+
+    def build(self):
+        copy(self, "fmiModel*.h", self.dependencies["fmi1"].cpp_info.components["modex"].includedirs[0],
+             os.path.join(self.source_folder, "src", "fmis", "FMI1"))
+        copy(self, "fmiPlatformTypes.h", self.dependencies["fmi1"].cpp_info.components["cosim"].includedirs[0],
+             os.path.join(self.source_folder, "src", "fmis", "FMI1"))
+        copy(self, "fmiFunctions.h", self.dependencies["fmi1"].cpp_info.components["cosim"].includedirs[0],
+             os.path.join(self.source_folder, "src", "fmis", "FMI1"))
+        copy(self, "*.h", self.dependencies["fmi2"].cpp_info.includedirs[0],
+             os.path.join(self.source_folder, "src", "fmis", "FMI2"))
+        copy(self, "*.h", self.dependencies["fmi3"].cpp_info.includedirs[0],
+             os.path.join(self.source_folder, "src", "fmis", "FMI3"))
+
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.md", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        copy(self, pattern="FMILIB_Acknowledgements.txt",
+             dst=os.path.join(self.package_folder, "licenses"),
+             src=self.source_folder)
+        copy(self, pattern="*.fmu", dst=os.path.join(self.package_folder, "res", "fmus"),
+             src=os.path.join(self.build_folder, "Testing"), keep_path=False)
+
+        cmake = CMake(self)
+        cmake.install()
```

Comment:
This is also installing `FMI1`, `FMI2`, and `FMI3` to the include folder, which as far as I can see, it's making those transitive headers public.

It's probably ok as-is, but Conan supports adding `transitive_headers=True` to the requirement so that it can be propagated to consumers, but I guess it would work as-is

Reply:
Since `fmilib` installs FMI headers to folders `FMI{1,2,3}`, using `transitive_headers=True` would also require manipulating the target include directories of these dependencies by prefixing them with `FMI{1,2,3}` to ensure that they are found. I'm not sure if this is possible if they aren't really in such a subfolder. `fmilib` uses a structure like `FMI{1,2,3}/*.h` in its public header files.

Reply:
Oh, I see, so maybe the current approach is the better way, thanks for the clarification!